src/main/java/net/minecraftforge/registries/DeferredRegister.java

```diff
@@ -152,41 +118,30 @@ public static <B> DeferredRegister<B> createOptional(ResourceLocation registryNa
     private SetMultimap<TagKey<T>, Supplier<T>> optionalTags;
     private boolean seenRegisterEvent = false;
 
-    private DeferredRegister(ResourceKey<? extends Registry<T>> registryKey, String modid, boolean optionalRegistry)
+    private DeferredRegister(ResourceKey<? extends Registry<T>> registryKey, String modid)
```

Comment:
Suggestion: Make this `protected` such that people will actually be able to extend `DeferredRegister`? I think this could be a nice goal; I know there are quite a few mods that already wrap `DeferredRegister`s to add their own custom util and have to effectively duplicate everything.